### PR TITLE
Setup Stripe and BE: Pay Route to test the setup

### LINKED ISSUE #5: BE: Pay Route
POST /request/:id/pay : Pay pet sitter for this request (service completed)

### LINKED ISSUE #40: Setup: Stripe API
<https://stripe.com/docs/api/customers/create>

<https://stripe.com/docs/payments/checkout/subscriptions>

Set up a dev account and use the appropriate API keys set up for the app.

### PR DESCRIPTION
**Pull request by samsoncsyu7777**
### What this PR does (required):
- Stripe single payment setup
- Stripe create customer setup
- Stripe subscription one time payment setup
- Setup 3 routes
- Frontend create Checkout Component, Checkout Form and Checkout route to test the APIs

### Screenshots / Videos (front-end only):
Screenshot 2021-11-20 at 4 50 31 PM

Screenshot 2021-11-20 at 4 51 39 PM

### Any information needed to test this feature (required):

BE: npm run dev
FE: npm start

### CODE REVIEW

File: server/controllers/payment.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
+const upload = require("../services/ImageUpload");
+const StripeServices = require("../services/StripePayment");
+const asyncHandler = require("express-async-handler");
+const { stripePayment, createCustomer, subscriptionOneTimePayment } = StripeServices;
+
+// @route POST /:id/single-pay
+// @desc create single payment checkout session
+// @access Private
```
Comment (sundayezeilo) [inline server/controllers/payment.js:8]: Good documentation

File: server/controllers/payment.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      res
+        .status(200)
+        .send({ message: "The payment session has been successfully created" });
+    });
+  }
+});
+
+// @route POST /:id/create-customer
+// @desc create customer
+// @access Private
```
Comment (sundayezeilo) [inline server/controllers/payment.js:33]: :+1:

File: server/controllers/payment.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+      res
+        .status(200)
+        .send({ message: "The customer has been successfully created", customer: req.createdCustomer });
+    });
+  }
+});
+
+// @route POST /:id/create-checkout-session
+// @desc create subscription one time payment checkout session
+// @access Private
```
Comment (sundayezeilo) [inline server/controllers/payment.js:58]: :+1:

**Review by sundayezeilo — COMMENTED** [review]
Looks good, @samsoncsyu7777. Resolve the merge conflicts, please. :smiley:

**Comment by samsoncsyu7777** [comment]
> Looks good, @samsoncsyu7777. Resolve the merge conflicts, please. 😃

OK. Done.

File: server/controllers/payment.js (hunk 1 above)
Comment (samsoncsyu7777) [reply server/controllers/payment.js:8]: Thank you.

File: server/controllers/payment.js (hunk 3 above)
Comment (samsoncsyu7777) [reply server/controllers/payment.js:58]: Thanks.

File: server/controllers/payment.js (hunk 2 above)
Comment (samsoncsyu7777) [reply server/controllers/payment.js:33]: :)

File: client/src/components/Checkout/Checkout.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
+import CssBaseline from '@material-ui/core/CssBaseline';
```
Comment (rajivtitus) [inline client/src/components/Checkout/Checkout.tsx:1]: Please remove unused import :)

File: client/src/components/Checkout/useStyles.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+const useStyles = makeStyles(() => ({
+  root: {
+    minHeight: '100vh',
+    '& .MuiInput-underline:before': {
+      borderBottom: '1.2px solid rgba(0, 0, 0, 0.2)',
+    },
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  authWrapper: {
```
Comment (rajivtitus) [inline client/src/components/Checkout/useStyles.ts:12]: I don't think we are using this CSS class, can we please remove it?

**Review by rajivtitus — APPROVED** [review]
Impressive work, I love the fact that you've created a temp UI to test the routes! 🚀 
Just a couple of minor comments to address before merging 👍🏽 

Important Note: Please remove the API keys from the PR description template as this is sensitive information (this is a public repo)

File: client/src/components/Checkout/Checkout.tsx (hunk 4 above)
Comment (samsoncsyu7777) [reply client/src/components/Checkout/Checkout.tsx:1]: Thank you. Done.

File: client/src/components/Checkout/useStyles.ts (hunk 5 above)
Comment (samsoncsyu7777) [reply client/src/components/Checkout/useStyles.ts:12]: Sure.

**Comment by samsoncsyu7777** [comment]
> Impressive work, I love the fact that you've created a temp UI to test the routes! 🚀 Just a couple of minor comments to address before merging 👍🏽
> 
> Important Note: Please remove the API keys from the PR description template as this is sensitive information (this is a public repo)

Thank you for reminding me.

**Pull request merged**